### PR TITLE
Create developer documentation for Micrometer

### PR DESCRIPTION
**Pull request by marcgs**
Developer documentation for instrumentation with Micrometer.

### CODE REVIEW

File: core/micrometer/README.md [hunk 1]
```diff
@@ -1,12 +0,0 @@
-# EDC extension for instrumentation with Micrometer
```
Comment (algattik) [inline core/micrometer/README.md:1]: nit: leave this file with a pointer to the other?

File: core/micrometer/README.md [hunk 2]
```diff
@@ -1,12 +0,0 @@
...
-
-This extension uses the [Micrometer](https://micrometer.io/) metrics library to automatically collect metrics from the host system, JVM, and frameworks and libraries used in EDC (including OkHttp and ExecutorService).
-
-## Configuration
-
-The following properties can use used to configure the metrics that this extension will collect:
```
Comment (algattik) [inline core/micrometer/README.md:7]: state that the value "false" should be used

File: docs/metrics.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+This extension provides support for instrumentation for the [Jetty](https://www.eclipse.org/jetty/) web server, which is enabled when using the `JettyExtension`.
+
+## Jersey Micrometer Extension
+
+This extension provides support for instrumentation for the [Jersey](https://eclipse-ee4j.github.io/jersey/) framework, which is enabled when using the `JerseyExtension`.
+
+## Configuration
+
+The following properties can use used to configure which metrics will be collected (default values are always "true"):
```
Comment (algattik) [inline docs/metrics.md:33]: state that the value "false" should be used

File: docs/metrics.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+EDC provides extensions for instrumentation with the [Micrometer](https://micrometer.io/) metrics library to automatically collect metrics from the host system, JVM, and frameworks and libraries used in EDC (including OkHttp, Jetty, Jersey and ExecutorService).
+
+## Micrometer Extension
+
+This extension provides support for instrumentation for some core EDC components:
+- JVM metrics
+- [OkHttp](https://square.github.io/okhttp/) client metrics
+- [ExecutorService](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/ExecutorService.html) metrics
+
+### Instrumenting ExecutorServices
```
Comment (algattik) [inline docs/metrics.md:12]: make clearer that this is independent of micrometer, and should be used in all cases.

and that a no op instrumentation is used in micrometer is not used.

File: docs/metrics.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
+# EDC extensions for instrumentation with Micrometer
+
+EDC provides extensions for instrumentation with the [Micrometer](https://micrometer.io/) metrics library to automatically collect metrics from the host system, JVM, and frameworks and libraries used in EDC (including OkHttp, Jetty, Jersey and ExecutorService).
+
+## Micrometer Extension
```
Comment (algattik) [inline docs/metrics.md:5]: add pointer to sample?

**Review by algattik — APPROVED** [review]

**Review by cpeeyush — APPROVED** [review]

**Review by ouphi — APPROVED** [review]

**Pull request merged**